packages/medusa/src/api/routes/store/carts/add-shipping-method.ts

```diff
@@ -67,24 +69,47 @@ export default async (req, res) => {
   const manager: EntityManager = req.scope.resolve("manager")
   const cartService: CartService = req.scope.resolve("cartService")
 
-  await manager.transaction(async (m) => {
-    const txCartService = cartService.withTransaction(m)
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
 
-    await txCartService.addShippingMethod(
-      id,
-      validated.option_id,
-      validated.data
-    )
+  const isWorkflowEnabled = featureFlagRouter.isFeatureEnabled({
+    workflows: Workflows.AddShippingMethod,
```

Comment:
question: what is the mechanism for turning these feature flags on?

Does the user do:
```
// medusa-config.js

... 
  featureFlags: {
    workflows: [ /* workflows to turn on here */ ]
  }
```

Reply:
Yep!

You can either do:
```
featureFlags: {
    workflows: {
      "add-shipping-method": true | false
    }
  }
```
or toggle all with:
```
featureFlags: {
    workflows: true | false
  }
```

And `"add-shipping-method"` can be an imported key from `@medusajs/workflows`.

See an example [here](https://github.com/medusajs/medusa/blob/1e7dbf8367dcd6bf890f1d3b316d0e11e03cfca8/integration-tests/plugins/medusa-config.js#L35).